ansible-k8s-vsphere/templates/user-data.j2

```diff
@@ -0,0 +1,57 @@
+#cloud-config
+
+users:
+# Admin User
+- name: {{ admin_ssh_user }}
+  primary_group: {{ admin_ssh_user }}
+  groups: users
+  sudo: ALL=(ALL) NOPASSWD:ALL
+  shell: /bin/bash
+  ssh_authorized_keys:
+  - {{ admin_ssh_public_key }}
+# Cluster User
+- name: {{ rke_ssh_user }}
+  primary_group: {{ rke_ssh_user }}
+  groups: docker
+  sudo: false
+  shell: /bin/bash
+  ssh_authorized_keys:
+  - {{ rke_ssh_public_key }} 
+
+# Update apt cache and dist-upgrade
+package_update: true
+package_upgrade: true
+
+packages:
+- open-vm-tools
+
+write_files:
+# Docker config
+- content: |
+    {
+      "exec-opts": ["native.cgroupdriver=systemd"],
+      "log-driver": "json-file",
+      "log-opts": {
+        "max-size": "100m"
+      },
+      "storage-driver": "overlay2"
+    }
+  path: /etc/docker/daemon.json
+
+runcmd:
+# Install Docker
+- curl -fsSL https://get.docker.com | sh
```

Comment:
From: https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script
> Docker provides convenience scripts at get.docker.com and test.docker.com for installing edge and testing versions of Docker Engine - Community into development environments quickly and non-interactively. The source code for the scripts is in the docker-install repository. **Using these scripts is not recommended for production environments, and you should understand the potential risks before you use them**

In the future, I think we can switch this out to use the recommended installation method of using the docker apt repositories. ([Installation Methods](https://docs.docker.com/engine/install/ubuntu/#installation-methods))

Reply:
resolved in 63f0e5db57a904e982f4bfac7686a9256d6340c7